app/main.py

```diff
@@ -1,25 +1,76 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self._name = name
+
+    def __get__(self, instance: Any, owner: Any) -> Any:
+        if instance is None:
+            return self
+        return instance.__dict__.get(self._name)
+
+    def __set__(self, instance: Any, value: Any) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Value must be an integer")
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError("Value out of range")
+        setattr(instance, self._name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
```

Comment:
The `Visitor` class should use the `IntegerRange` descriptors for `age`, `weight`, and `height` to ensure that these attributes are validated upon assignment. This will ensure that the visitor's attributes are always within the specified range.